### PR TITLE
tui: use xterm focus reporting escape sequence for urxvt 9.23+

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/neovim/neovim/issues/8695#issuecomment-403108752

Needed lint adjustments.. :/

### CODE REVIEW

File: src/nvim/tui/tui.c [hunk 1]
```diff
@@ -1722,30 +1722,25 @@ static void augment_terminfo(TUIData *data, const char *term,
...
+  data->unibi_ext.enable_focus_reporting = (int)unibi_add_ext_str(
+      ut, "ext.enable_focus",
+      rxvt ? "\x1b[?1004h\x1b]777;focus;on\x7" : "\x1b[?1004h");
+  data->unibi_ext.disable_focus_reporting = (int)unibi_add_ext_str(
+      ut, "ext.disable_focus",
+      rxvt ? "\x1b[?1004l\x1b]777;focus;off\x7" : "\x1b[?1004l");
+  data->unibi_ext.enable_mouse = (int)unibi_add_ext_str(
+      ut, "ext.enable_mouse", "\x1b[?1002h\x1b[?1006h");
+  data->unibi_ext.disable_mouse = (int)unibi_add_ext_str(
+      ut, "ext.disable_mouse", "\x1b[?1002l\x1b[?1006l");
```
Comment (blueyed) [inline src/nvim/tui/tui.c]: Re formatting: I could also put `ut` on a seperate line (smaller diff), but I think this is OK to be compact?!

**Comment by justinmk** [comment]
can you put the full explanation from https://github.com/neovim/neovim/issues/8695 in the commit message?

File: src/nvim/tui/tui.c [hunk 2]
```diff
@@ -1722,30 +1722,25 @@ static void augment_terminfo(TUIData *data, const char *term,
...
-      "\x1b[?1002l\x1b[?1006l");
+  data->unibi_ext.enable_lr_margin = (int)unibi_add_ext_str(
+      ut, "ext.enable_lr_margin", "\x1b[?69h");
+  data->unibi_ext.disable_lr_margin = (int)unibi_add_ext_str(
+      ut, "ext.disable_lr_margin", "\x1b[?69l");
+  data->unibi_ext.enable_bracketed_paste = (int)unibi_add_ext_str(
+      ut, "ext.enable_bpaste", "\x1b[?2004h");
+  data->unibi_ext.disable_bracketed_paste = (int)unibi_add_ext_str(
+      ut, "ext.disable_bpaste", "\x1b[?2004l");
+  /// Use custom focus reporting code for urxvt extension (up to urxvt 9.22).
```
Comment (justinmk) [inline src/nvim/tui/tui.c]: I was thinking more like:

    // For urxvt send BOTH xterm and old urxvt sequences. #8695

**Comment by blueyed** [comment]
Done.

**Pull request merged**